main.cpp

```diff
@@ -79,132 +21,31 @@ int usage(const char* argv0, const int ret)
-	}
-	if(outDir.isEmpty())
-	{
-		std::cerr << "Output directory not specified.\n";
-		return usage(argv[0], 1);
-	}
-	if(poDir.isEmpty())
-		std::cerr << "Warning: translations (po) directory not specified. Will not load existing translations of names.\n";
-
-	if(QFile(outDir).exists())
-	{
-		std::cerr << "Output directory already exists, won't touch it.\n";
-		return 1;
-	}
-	if (!QFile(inDir+"/info.ini").exists())
-	{
-		std::cerr << "Error: info.ini file wasn't found\n";
-		return 1;
-	}
-	inDir = QDir::fromNativeSeparators(inDir);
-	// Remove trailing directory separators, so that QFileInfo::fileName
-	// gave sky culture id rather than an empty string.
-	while(inDir.endsWith("/")) inDir.chop(1);
-	std::stringstream out;
-	QString boundariesType, author, credit, license, cultureId, region, englishName;
-	convertInfoIni(inDir, out, boundariesType, author, credit, license, cultureId, region, englishName);
-
-	AsterismOldLoader aLoader;
-	aLoader.load(inDir, cultureId);
-
-	ConstellationOldLoader cLoader;
-	cLoader.setBoundariesType(boundariesType.toStdString());
-	cLoader.load(inDir, outDir, nativeLocale);
-
-	NamesOldLoader nLoader;
-	nLoader.load(inDir, nativeLocale, convertUntranslatableNamesToNative);
-
-	std::cerr << "Starting emission of JSON...\n\n";
-	aLoader.dumpJSON(out);
-	cLoader.dumpJSON(out);
-	nLoader.dumpJSON(out);
-
-	auto str = std::move(out).str();
-	writeEnding(str);
-
-	if(!QDir().mkpath(outDir))
-	{
-		std::cerr << "Failed to create output directory\n";
-		return 1;
-	}
-	{
-		std::ofstream outFile((outDir+"/index.json").toStdString());
-		outFile << str;
-		outFile.flush();
-		if(!outFile)
-		{
-			std::cerr << "Failed to write index.json\n";
-			return 1;
-		}
-	}
-
-	DescriptionOldLoader dLoader;
-	license = convertLicense(license);
-	dLoader.load(inDir, poDir, cultureId, englishName, author, credit, license, cLoader, aLoader, nLoader,
-	             footnotesToRefs, genTranslatedMD);
-	dLoader.dump(outDir);
-
-	std::cerr << "--- NOTE ---\n";
-	if(region == "???")
-	{
-		std::cerr << "* Some JSON values can't be deduced from the old-format data. They have been"
-			     " marked by \"???\". Please replace them with something sensible.\n";
-	}
-	std::cerr << "* The key \"langs_use_native_names\" is omitted since it has no counterpart"
-	             " in the old format. If this sky culture needs it, please add it manually.\n";
-	std::cerr << "* The transformation of the description text is very basic, please check that"
-	             " it looks as it should. Pay special attention at References, Authors, and"
-	             " License sections, which may have been formulated in a suboptimal way.\n";
+    QCoreApplication app(argc, argv);
+    QString inDir, outDir, poDir, nativeLocale;
+    bool footnotesToRefs = false, genTranslatedMD = false, convertUntranslatableNamesToNative = false;
+    // parse arguments
+    std::vector<std::string> args(argv+1, argv+argc);
+    for(const auto& arg : args) {
```

Comment:
Please use the coding style described [here](https://stellarium.org/doc/25.0/codingStyle.html).